whois-query/src/main/java/net/ripe/db/whois/query/hazelcast/HazelcastInstanceManager.java

```diff
@@ -14,42 +14,67 @@
 import org.springframework.context.annotation.Primary;
 import org.springframework.context.annotation.Profile;
 
+import java.util.Arrays;
+
 @Configuration
 @DeployedProfile
 public class HazelcastInstanceManager {
     private static final Logger LOGGER = LoggerFactory.getLogger(HazelcastInstanceManager.class);
 
     final String interfaces;
+    final int port;
 
     @Autowired
-    public HazelcastInstanceManager( @Value("${hazelcast.config.interfaces}") final String interfaces) {
+    public HazelcastInstanceManager(@Value("${hazelcast.config.interfaces}") final String interfaces, @Value("${hazelcast.port:5701}") final int port) {
         this.interfaces = interfaces;
+        this.port = port;
     }
 
     @Bean
     @Profile(WhoisProfile.RIPE_DEPLOYED)
     public HazelcastInstance hazelcastInstance() {
       LOGGER.info("Creating hazelcast instance with Ripe deployed profile");
-      final HazelcastInstance instance =  Hazelcast.newHazelcastInstance(null);
-      instance.getCluster().addMembershipListener(new HazelcastMemberShipListener());
 
-      return instance;
+      final Config config = getGenericConfig();
+      //We define ipv6 addresses
+      config.setProperty("hazelcast.prefer.ipv4.stack", "false");
+
+      config.getNetworkConfig().getJoin().getAwsConfig().setEnabled(false);
+      config.getNetworkConfig().setPort(port).setPortAutoIncrement(false);
+      config.getNetworkConfig().getJoin().getTcpIpConfig().setMembers(Arrays.asList(interfaces.split(","))).setEnabled(true);
+
+      return getHazelcastInstance(config);
     }
 
     @Bean
     @Profile(WhoisProfile.AWS_DEPLOYED)
     @Primary
     public HazelcastInstance hazelcastAwsInstance() {
-        LOGGER.info("Creating hazelcast instance with AWS deployed profile, interfaces {}", interfaces);
+        LOGGER.info("Creating hazelcast instance with AWS deployed profile");
 
-        final Config config = new Config();
-        config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
+        final Config config = getGenericConfig();
```

Comment:
Is Multicast config disabled by default?

Reply:
its enabled by default. We need to disable if using any other means